nemo/collections/speechlm/data/text_processing.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from lhotse.cut import Cut
+
+from nemo.collections.multimodal.speech_llm.parts.utils.data_utils import PromptFormatterTextProcessing
+
+
+__all__ = ['TextProcesserWithPromptFormatter']
+
+
+@dataclass
+class MockCutSupervision:
+    """
+    A dummy class for MockCut to support behavior of Cut.supervisions[0].text
+    """
+
+    text: str = ""
+
+
+class MockCut(Cut):
```

Comment:
What is this? It will only work with a small subset of PromptFormatters for which `@registered_prompt_format_fn` reads from `context` / `question` / `answer`.

Reply:
This shouldn't be needed, please add a data type following `nemo/collections/common/data/lhotse/text_adapters.py` and register a prompt format fn for it. You can follow the tutorial at https://github.com/NVIDIA/NeMo/blob/main/tutorials/multimodal/Prompt%20Formatter%20Tutorial.ipynb

Reply:
This is necessary for using the prompt formatter in both lhotse and non-lhotse datasets. It's not creating a new prompt format or adapter.